library/src/scala/util/FromString.scala

```diff
@@ -0,0 +1,47 @@
+package scala.util
+
+trait FromString[T] {
```

Comment:
`Option` is clearly not an adequate replacement, since it does not allow to pass an error string _why_ the argument does not conform. 

Also, in the end my user-level method takes, say an `Int` and `Boolean`. If the arguments do not conform to that I _must_ throw an exception. There's no other way to do it. Unless you make @main programs take only `Option` or `Either` arguments, but that's completely wrong! The whole point of @main functions is that the person writing the @main can decide what degree of error checking should be done, and how. If you force @main functions to handle errors by analyzing `Option` or `Either` arguments you made the choice for them.

In summary: exceptions are actually pretty nice! And we will fix the part that they don't show up in the types...